tests/devices/i04/test_transfocator.py

```diff
@@ -1,66 +1,75 @@
-from unittest.mock import MagicMock, patch
+import asyncio
+from unittest.mock import AsyncMock, patch
 
 import pytest
-from ophyd.sim import make_fake_device
-from ophyd.status import SubscriptionStatus
+from ophyd_async.core import (
+    DeviceCollector,
+    set_mock_value,
+    wait_for_value,
+)
 
 from dodal.devices.i04.transfocator import Transfocator
 
 
 @pytest.fixture
-def fake_transfocator():
-    FakeTransfocator = make_fake_device(Transfocator)
-    transfocator: Transfocator = FakeTransfocator(name="test_transfocator")
-    yield transfocator
+async def fake_transfocator() -> Transfocator:
+    async with DeviceCollector(mock=True):
+        transfocator = Transfocator(prefix="", name="transfocator")
+    return transfocator
 
 
 def given_predicted_lenses_is_half_of_beamsize(transfocator: Transfocator):
     def lens_number_is_half_beamsize(value, *args, **kwargs):
-        transfocator.predicted_vertical_num_lenses.sim_put(int(value / 2))  # type: ignore
+        set_mock_value(transfocator.predicted_vertical_num_lenses, int(value / 2))
 
-    transfocator.beamsize_set_microns.subscribe(lens_number_is_half_beamsize)
+    transfocator.beamsize_set_microns.subscribe_value(lens_number_is_half_beamsize)
 
 
-def test_given_beamsize_already_set_then_when_transfocator_set_then_returns_immediately(
+async def set_beamsize_to_same_value_as_mock_signal(
```

Comment:
I think the functions here are in the wrong order. In the test below, we want to check that doing the set returns immediately AFTER `beamsize_set_microns` has been set - not before.